src/Xunit.StaFact/Sdk/UITestRunner.cs

```diff
@@ -19,4 +19,37 @@ protected override Task<decimal> InvokeTestMethodAsync(ExceptionAggregator aggre
     {
         return new UITestInvoker(this.Test, this.MessageBus, this.TestClass, this.ConstructorArguments, this.TestMethod, this.TestMethodArguments, this.BeforeAfterAttributes, aggregator, this.CancellationTokenSource, this.threadRental).RunAsync();
     }
+
+    private sealed class FilteringMessageBus : IMessageBus
+    {
+        private readonly IMessageBus messageBus;
+
+        public FilteringMessageBus(IMessageBus messageBus)
+        {
+            this.messageBus = messageBus;
+        }
+
+        public void Dispose()
+        {
+            this.messageBus.Dispose();
+        }
+
+        public bool QueueMessage(IMessageSinkMessage message)
+        {
+            if (message is ITestFailed testFailed)
+            {
+                // This test will run again; report it as skipped instead of failed
+                // TODO: What kind of additional logs should we include?
+                message = new TestSkipped(testFailed.Test, "Test will automatically retry.");
+            }
+            else if (message is ITestCleanupFailure testCleanupFailure)
+            {
+                // This test will run again; report it as skipped instead of failed
+                // TODO: What kind of additional logs should we include?
+                message = new TestSkipped(testCleanupFailure.Test, "Test will automatically retry.");
+            }
```

Comment:
📝 vs-extension-testing doesn't handle this case. I'm not sure if that was intentional or an oversight.